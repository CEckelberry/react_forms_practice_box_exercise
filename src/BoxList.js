import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
    const INITIAL_STATE = [
        {id: "598b53d4-7513-4ee1-92a8-d7fe9830fb15", backgroundColor: "green", height: 100, width: 100}
    ]

    const [boxes, setBoxes] = useState(INITIAL_STATE);
    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, {...newBox, id: uuid() }]);
    }

    const remove = (id) => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
    }

    return (
        <div>
            <h1>Boxes Emporium!</h1>
            <NewBoxForm addBox={addBox} />
            <div>
                {boxes.map(({ id, backgroundColor, width, height }) => <Box id={id} backgroundColor={backgroundColor} width={width} height={height} handleRemove={remove}/>)}
            </div>
        </div>
    )
}

export default BoxList;