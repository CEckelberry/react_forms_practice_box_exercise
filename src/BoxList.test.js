import React from "react";
import { render, fireEvent, getByTitle, screen } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", () => {
    render(<BoxList />);
});

it("matches the snapshot", () => {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add a new box", () => {
    const { queryByText, getByLabelText } = render(<BoxList />);
    const colorInput = getByLabelText("backgroundColor:");
    const widthInput = getByLabelText("width:");
    const heightInput = getByLabelText("height:");
    const btn = queryByText("Add Box");
    
    expect(queryByText('style="background-color: yellow')).not.toBeInTheDocument();
    expect(queryByText('backgroundColor:')).toBeInTheDocument();
    fireEvent.change(colorInput, { target: {value: 'yellow'}});
    fireEvent.change(widthInput, { target: {value: '150'}});
    fireEvent.change(heightInput, { target: {value: '150'}});
    fireEvent.click(btn);
    const yellowBox = screen.getByTitle('yellow')
    expect(yellowBox).toBeInTheDocument();
})