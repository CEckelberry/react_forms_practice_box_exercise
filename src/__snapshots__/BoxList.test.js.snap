// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Boxes Emporium!
    </h1>
    <form>
      <label
        for="backgroundColor"
      >
        backgroundColor: 
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="background color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        width: 
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        height: 
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div>
      <div>
        <div
          id="598b53d4-7513-4ee1-92a8-d7fe9830fb15"
          style="background-color: green; width: 100px; height: 100px;"
          title="green"
        />
        <button>
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
